framework/web/Request.php

```diff
@@ -913,15 +989,33 @@ public function getReferrer()
      */
     public function getUserAgent()
     {
-        return isset($_SERVER['HTTP_USER_AGENT']) ? $_SERVER['HTTP_USER_AGENT'] : null;
+        return $this->headers->get('User-Agent');
     }
 
     /**
      * Returns the user IP address.
+     * The IP is determined using headers and / or `$_SERVER` variables.
+     * @since 2.0.13
      * @return string|null user IP address, null if not available
      */
     public function getUserIP()
     {
+        foreach ($this->ipHeaders as $ipHeader) {
+            if ($this->headers->has($ipHeader)) {
+                return trim(explode(',', $this->headers->get($ipHeader))[0]);
+            }
+        }
+        return $this->getRemoteIP();
```

Comment:
Only if the host is trusted.
The comment in the source code regarding the default value is out of date. Will fix that.